docs/source/tutorials/using_text.rst

```diff
@@ -0,0 +1,461 @@
+        def construct(self):
+            t = Text("Hello", gradient=(RED, BLUE, GREEN)).scale(2)
+            self.add(t)
+
+You can also use :attr:`~.Text.t2g` for gradients with specific 
+characters of the text. It shares a similar syntax to :ref:`Using Colors`:
+
+.. manim:: t2gExample
+    :save_last_frame:
+
+    class t2gExample(Scene):
+        def construct(self):
+            t2gindices = Text(
+                'Hello',
+                t2g={
+                    '[1:-1]': (RED,GREEN),
+                },
+            ).move_to(LEFT)
+            t2gwords = Text(
+                'World',
+                t2g={
+                    'World':(RED,BLUE),
+                },
+            ).next_to(t2gindices, RIGHT)
+            self.add(t2gindices, t2gwords)
+
+Setting Line Spacing
+--------------------
+
+You can set line spacing using :attr:`~.Text.line_spacing`:
+
+.. manim:: LineSpacing
+    :save_last_frame:
+
+    class LineSpacing(Scene):
+        def construct(self):
+            a = Text("Hello\nWorld", line_spacing=1)
+            b = Text("Hello\nWorld", line_spacing=4)
+            self.add(Group(a,b).arrange(LEFT, buff=5))
+
+
+.. _disable-ligatures:
+
+Disabling Ligatures
+-------------------
+
+By disabling ligatures you would get a one-to-one mapping between characters and
+submobjects. This fixes the issues with coloring text. 
+
+
+.. warning::
+
+    Be aware that using this method with text that heavily depends on
+    ligatures (Arabic text) may yield unexpected results.
+
+You can disable ligatures by passing ``disable_ligatures`` to 
+:class:`Text`. For example:
+
+.. manim:: DisableLigature
+    :save_last_frame:
+
+    class DisableLigature(Scene):
+        def construct(self):
+            li = Text("fl ligature").scale(2)
+            nli = Text("fl ligature", disable_ligatures=True).scale(2)
+            self.add(Group(li, nli).arrange(DOWN, buff=.8))
+
+.. _iterating-text:
+
+Iterating :class:`~.Text`
+-------------------------
+
+Text objects behave like :class`~.VGroup`s. Therefore, you can slice and index
+the text.
+
+For example, you can set each letter to different color by iterating it.
+
+.. manim:: IterateColor
+    :save_last_frame:
+
+    class IterateColor(Scene):
+        def construct(self):
+            text = Text("Colors").scale(2)
+            for letter in text:
+                letter.set_color(random_bright_color())
+            self.add(text)
+
+.. warning::
+
+    Please note that `Ligature`_ can cause problems here. If you need a
+    one-to-one mapping of characters to submobjects you should pass
+    the ``disable_ligatures`` parameter to :class:`~.Text`.
+    See :ref:`disable-ligatures`
+
+.. _Ligature: https://en.wikipedia.org/wiki/Ligature_(writing)
+
+Working with :class:`~.MarkupText`
+==================================
+
+MarkupText is similar to :class:`~.Text`, the only difference between them is 
+that this accepts PangoMarkup (which is similar to html), instead of text.
```

Comment:
Maybe be a bit more explicit here?

```suggestion
that this accepts and processes PangoMarkup (which is similar to
html), instead of just rendering plain text.
```